Invalid suggestion for parenthesized break

### Code

```Rust
fn main() {
    {
        (break)
    };
}
```

### Current output

```Shell
warning: unnecessary parentheses around block return value
 --> src/main.rs:3:9
  |
3 |         (break)
  |         ^     ^
  |
  = note: `#[warn(unused_parens)]` (part of `#[warn(unused)]`) on by default
help: remove these parentheses
  |
3 -         (break)
3 +         break 
  |

error[E0268]: `break` outside of a loop or labeled block
 --> src/main.rs:3:9
  |
3 |         (break)
  |         ^^^^^^^ cannot `break` outside of a loop or labeled block
  |
help: consider labeling this block to be able to break within it
  |
2 ~     'block: {
3 ~         (brea 'blockk)
  |

For more information about this error, try `rustc --explain E0268`.
warning: `playground` (bin "playground") generated 1 warning
```

### Desired output

```Shell

```

### Rationale and extra context

`brea 'blockk` is invalid

### Other cases

```Rust

```

### Rust Version

```Shell
Reproducible on the playground with version ` 1.97.0-nightly (2026-05-09 82bee965077a631d6fbd)`
```

### Anything else?

_No response_

<!-- TRIAGEBOT_START -->

<!-- TRIAGEBOT_ASSIGN_START -->

<!-- TRIAGEBOT_ASSIGN_END -->
<!-- TRIAGEBOT_END -->

--- a/compiler/rustc_hir_typeck/src/loops.rs
+++ b/compiler/rustc_hir_typeck/src/loops.rs
@@ -270,7 +270,13 @@ impl<'hir> Visitor<'hir> for CheckLoopVisitor<'hir> {
                     }
                 }
 
-                let sp_lo = e.span.with_lo(e.span.lo() + BytePos("break".len() as u32));
+                let sp_lo = if let Ok(snippet) = self.tcx.sess.source_map().span_to_snippet(e.span)
+                    && let Some(break_pos) = snippet.find("break")
+                {
+                    e.span.with_lo(e.span.lo() + BytePos((break_pos + "break".len()) as u32))
+                } else {
+                    e.span.with_lo(e.span.lo() + BytePos("break".len() as u32))
+                };
                 let label_sp = match break_destination.label {
                     Some(label) => sp_lo.with_hi(label.ident.span.hi()),
                     None => sp_lo.shrink_to_lo(),

--- a/tests/ui/parser/break-in-unlabeled-block-parenthesized.rs
+++ b/tests/ui/parser/break-in-unlabeled-block-parenthesized.rs
@@ -1,0 +1,9 @@
+#![allow(unused_parens)]
+fn main() {
+    {
+        (break); //~ ERROR `break` outside of a loop or labeled block
+    };
+    {
+        ((break)); //~ ERROR `break` outside of a loop or labeled block
+    };
+}

--- a/tests/ui/parser/break-in-unlabeled-block-parenthesized.stderr
+++ b/tests/ui/parser/break-in-unlabeled-block-parenthesized.stderr
@@ -1,0 +1,27 @@
+error[E0268]: `break` outside of a loop or labeled block
+  --> $DIR/break-in-unlabeled-block-parenthesized.rs:4:9
+   |
+LL |         (break);
+   |         ^^^^^^^ cannot `break` outside of a loop or labeled block
+   |
+help: consider labeling this block to be able to break within it
+   |
+LL ~     'block: {
+LL ~         (break 'block);
+   |
+
+error[E0268]: `break` outside of a loop or labeled block
+  --> $DIR/break-in-unlabeled-block-parenthesized.rs:7:9
+   |
+LL |         ((break));
+   |         ^^^^^^^^^ cannot `break` outside of a loop or labeled block
+   |
+help: consider labeling this block to be able to break within it
+   |
+LL ~     'block: {
+LL ~         ((break 'block));
+   |
+
+error: aborting due to 2 previous errors
+
+For more information about this error, try `rustc --explain E0268`.

